database/migrations/2017_04_27_183942_drop_unique_northstar_id_index_reactions_table.php

```diff
@@ -13,7 +13,7 @@ class DropUniqueNorthstarIdIndexReactionsTable extends Migration
     public function up()
     {
         Schema::table('reactions', function (Blueprint $table) {
-            $table->dropUnique('kudos_file_id_drupal_id_northstar_id_taxonomy_id_unique');
+            //
```

Comment:
Why is this removed?

Reply:
The index didn't exist, so it was unable to drop it.

Reply:
Sorry this is like time traveling and breaks my brain. Isn't it created in `2016_07_27_172344_create_kudos_table.php`?

Reply:
Alright yeah – it's created there. The problem that came up was that we deleted `file_id`, a part of that compound index, in [this migration](https://github.com/DoSomething/rogue/pull/500/files#diff-3d5477b189c52ceb8a7b12d8639cf21e) in 2016. When we originally were trying to recreate the index in the `down` migration here, we couldn't:

1. We're trying to just set an index by name... but not specifying what columns it should be tracking.
2. We wouldn't be able to recreate the index here since the `kudos_id` column was deleted back in 2016, and so wouldn't have been recreated yet (we're still "working back" from 2018).

Deleting the index [back then](https://github.com/DoSomething/rogue/pull/500/files#diff-3d5477b189c52ceb8a7b12d8639cf21e) seemed to be the most sensible call at this point.

Reply:
At least I think that's what's happening. 

![](https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/118/shocked-face-with-exploding-head_1f92f.png)

Reply:
@DFurnes Yes okay I'm onboard with that I think that is correct. You have merged the operations of `2017_04_27_183942_drop_unique_northstar_id_index_reactions_table.php` and `2016_10_11_200952_drop_file_id_from_reactions.php` into `2016_10_11_200952_drop_file_id_from_reactions.php`.

In that case should `2017_04_27_183942_drop_unique_northstar_id_index_reactions_table.php` be deleted since it is not doing anything?

Reply:
I'm nervous about deleting a migration, since our existing instances will have thought they'd run it (so if you ran `php artisan migrate:rollback` on your machine after pulling this, it'd explode if it couldn't find the file it thought it had run).

Reply:
Oh right right right yeah it's in the `migrations` table already. Okay!